version: '3'

services:
  db:
    image: postgres:10
    container_name: django_db
    environment:
      POSTGRES_USER: django
      POSTGRES_PASSWORD: django
    volumes:
      - ./data:/var/lib/postgresql/data
  web:
    build: .
    container_name: django
    command: python3 manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/code
    ports:
      - "8000:8000"
    depends_on:
      - db

  openldap:
    image: osixia/openldap:1.2.1
    container_name: openldap
    environment:
      LDAP_LOG_LEVEL: "256"
      LDAP_ORGANISATION: "FSEI"
      LDAP_DOMAIN: "openldap"
      LDAP_ADMIN_PASSWORD: "admin"
      LDAP_CONFIG_PASSWORD: "config"
    volumes:
      - ./openldap/database:/var/lib/ldap
      - ./openldap/conf:/etc/ldap/slapd.d
    ports:
      - "389:389"
      - "636:636"
    #domainname: "openldap" # bug
    hostname: "openldap"
    tty: true
    stdin_open: true
    command: --loglevel=debug

  phpldapadmin:
    image: osixia/phpldapadmin:0.7.1
    container_name: phpldapadmin
    environment:
      PHPLDAPADMIN_LDAP_HOSTS: "openldap"
      PHPLDAPADMIN_HTTPS: "false"
    volumes:
      - ./phpldapadmin/certs:/container/service/phpldapadmin/assets/apache2/certs
    ports:
      - "8001:80"
    depends_on:
      - openldap
